packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-swift-visitor.ts

```diff
@@ -170,16 +173,36 @@ export class AppSyncSwiftVisitor extends AppSyncModelVisitor {
   }
 
   private generateFieldSchema(field: CodeGenField, modelKeysName: string): string {
-    if (field.type === 'ID') {
+    if (field.type === 'ID' && field.name === 'id') {
```

Comment:
what happens if I call my ID field 'iidd'?

Reply:
There is a restriction in the `DataModel` to to have restriction where `id` field has to be present and should be of the type `ID`